chain/epoch-manager/src/shard_assignment.rs

```diff
@@ -342,100 +340,6 @@ pub(crate) fn assign_chunk_producers_to_shards(
     Ok(result)
 }
 
-pub(crate) mod old_validator_selection {
```

Comment:
You may want to check with @shreyan-gupta that this isn't needed for epoch sync

Reply:
I need to go back and take a detailed look at epoch sync and see what code can we delete and what code we need to keep. High level context, epoch sync is one of those things that we don't have a good coverage of, and along with genesis, it's something we need to always support older versions to construct a valid proof.